includes/admin/class-wcs-admin-post-types.php

```diff
@@ -660,14 +660,33 @@ public static function get_date_column_content( $subscription, $column ) {
 
 		$date_type_map = array( 'last_payment_date' => 'last_order_date_created' );
 		$date_type     = array_key_exists( $column, $date_type_map ) ? $date_type_map[ $column ] : $column;
+		$date_timestamp    = $subscription->get_time( $date_type, 'site' );
 
 		if ( 0 == $subscription->get_time( $date_type, 'gmt' ) ) {
 			$column_content = '-';
 		} else {
-			$column_content = sprintf( '<time class="%s" title="%s">%s</time>', esc_attr( $column ), esc_attr( date( __( 'Y/m/d g:i:s A', 'woocommerce-subscriptions' ), $subscription->get_time( $date_type, 'site' ) ) ), esc_html( $subscription->get_date_to_display( $date_type ) ) );
+			$column_content = sprintf( '<time class="%s" title="%s">%s</time>', esc_attr( $column ), esc_attr( date( __( 'Y/m/d g:i:s A', 'woocommerce-subscriptions' ), $date_timestamp ) ), esc_html( $subscription->get_date_to_display( $date_type ) ) );
```

Comment:
I'm getting a pesky phpcs error on this line. 

>  668 | ERROR   | [ ] date() is affected by runtime timezone changes which can cause date/time to be incorrectly displayed. Use gmdate() instead.
 
This is existing code – we're using `date()` already. This seems correct to me, we want to display the date in the server's timezone. I don't understand why this rule exists, and changing to `gmdate` would introduce a bug. 

Is there an alternative that allows us to explicitly request a localised date, e.g. something like `date( 'server_timezone' )`? Or do we want to turn this rule off?

Reply:
Ah yes. 

The phpcs error is technically correct, we should never use date functions that are affected by runtime timezone changes. We've had pretty bad subscription issues in the distant past because folks change their PHP timezone for some reason and we were using functions like `date()`.

But the suggestion phpcs gives to use `gmdate()` raises some confusing date-related questions, that don't make a tonne of sense. 

---

The php manual says that `gmdate()` is: _"Identical to the [date()](https://www.php.net/manual/en/function.date.php) function except that the time returned is Greenwich Mean Time (GMT)."_ 

But IMO that is kinda misleading because in my understanding it doesn't manipulate the timestamp input at all to make it refer the GMT timezone equivalent. Which is why phpcs is recommending it. 

So, to be clear, calling `gmdate( '...', $timestamp )` where the timestamp param has already been adjusted to reflect the site's timezone, will display a date that reflects the site's timezone, it won't a GMT date at all. 

I hope that makes sense, but its super confusing.

Reply:
Just to add more details.

We store subscription dates in the database in GMT (UTC+0). 
When you call `$subscription->get_time( ..., 'site' )` we take the date stored in GMT, convert it to a timestamp and then offset it based on the site's timezone offset. eg +10 hrs, -7 hrs etc. 

So for example: 

If I purchase a subscription at 1 pm brisbane time, the start date in the database is recorded as `2023-01-24 03:12:09` which is 13:12 AEST - 10 hrs = 03:12 UTC

If I then call `$subscription->get_time( ... )` I will get a timestamp that is 03:12. 

If I call `$subscription->get_time( ..., 'site' )` I will get a timestamp that is 13:12.

If we call `gmdate( ..., $subscription->get_time( ..., 'site' ) )` the date displayed will be 13:12 which is correct.

What we've done is passed it a site converted timestamp, it's basically the same as if we called `gmdate` with the timestamp equal to 13:12 in GMT.

I hope that makes sense, but to be really explicit, the phpcs is correct, we should just call `gmdate()` instead of `date()` here. It should be the same outcome.

Reply:
Great explanation! Makes sense, I was confused and thought that the localisation (timezone offset) provided by `gmdate` was valid, I see now that it doesn't rely on consistent settings. I guess there's a possibility this fixes a bug, some sites might have seen different dates.